src/dxtbx/format/FormatNXmxEigerFilewriter.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import re
+
+import h5py
+
+from scitbx.array_family import flex
+
+import dxtbx.nexus
+from dxtbx.format.FormatNXmx import FormatNXmx
+from dxtbx.nexus import _dataset_as_flex, get_detector_module_slices
+
+DATA_NNNNNN_RE = re.compile(r"data_\d{6}")
+
+
+class FormatNXmxEigerFilewriter(FormatNXmx):
+
+    _cached_file_handle = None
+
+    @staticmethod
+    def understand(image_file):
+        with h5py.File(image_file) as handle:
+            if "/entry/instrument/detector/detectorSpecific/eiger_fw_version" in handle:
+                return True
+        return False
+
+    def __init__(self, image_file, **kwargs):
+        """Initialise the image structure from the given file."""
+        super().__init__(image_file, **kwargs)
+
+    def _get_nxmx(self, fh: h5py.File):
+        nxmx = dxtbx.nexus.nxmx.NXmx(fh)
+        nxentry = nxmx.entries[0]
+
+        nxdetector = nxentry.instruments[0].detectors[0]
```

Comment:
`assert len(instruments) == 1` 

and 

`assert len(instruments[0].detectors) == 1`

?